internal/server/statvar/statvar_summary.go

```diff
@@ -56,8 +58,33 @@ func GetStatVarSummary(
 	result := &pb.GetStatVarSummaryResponse{
 		StatVarSummary: map[string]*pb.StatVarSummary{},
 	}
-	for dcid, data := range baseDataList[0] {
-		result.StatVarSummary[dcid] = data.(*pb.StatVarSummary)
+	// Merge strategy
+	// 1. "place_type_summary": For a given place type, pick the Bigtable with the
+	//    most places.
+	// 2. "provenance_summary": Merge provenances from all the Bigtables.
+	for _, baseData := range baseDataList {
+		for dcid, data := range baseData {
+			svs, ok := data.(*pb.StatVarSummary)
+			if !ok {
+				return nil, status.Errorf(codes.Internal, "Can not read StatVarSummary")
+			}
+			if _, ok := result.StatVarSummary[dcid]; !ok {
+				result.StatVarSummary[dcid] = svs
+				continue
+			}
+			// Pick place type summary with the most places.
+			for pt := range svs.PlaceTypeSummary {
+				summary, ok := result.StatVarSummary[dcid].PlaceTypeSummary[pt]
+				if ok && svs.PlaceTypeSummary[pt].NumPlaces < summary.NumPlaces {
+					continue
+				}
+				result.StatVarSummary[dcid].PlaceTypeSummary[pt] = svs.PlaceTypeSummary[pt]
+			}
+			//
+			for source := range svs.ProvenanceSummary {
+				result.StatVarSummary[dcid].ProvenanceSummary[source] = svs.ProvenanceSummary[source]
```

Comment:
if there is an overlapping source, maybe we could prefer the one from the better cache?

Reply:
Done